File: src/MrHotel/WebApp/src/app/services/reservation-manager/data/room.ts
import { Guid } from "@customTypes/.";
import { Tag } from ".";

export type Room = {
  id: Guid;
  name: string;
  tags: Tag[];
};

export function roomEmpty(id: Guid, name: string): Room {
  return {
    id: id,
    name: name,
    tags: [],
  };
}

File: src/MrHotel/Database/Entities/Reservations/Room.cs
namespace MrHotel.Database.Entities.Reservations;

public class Room
{
    public Guid Id { get; set; }

    public required string Name { get; set; }

    public ICollection<Tag> Tags { get; set; } = [];

    public static Room Create(string name)
    {
        return new()
        {
            Id = Guid.NewGuid(),
            Name = name,
        };
    }
}
